diff --git a/h5py/h5p.pyx b/h5py/h5p.pyx
index da175dd..45c7fa9 100644
--- a/h5py/h5p.pyx
+++ b/h5py/h5p.pyx
@@ -25,7 +25,7 @@ from h5py import _objects
 from ._objects import phil, with_phil
 
 if MPI:
-    from mpi4py.mpi_c cimport MPI_Comm, MPI_Info, MPI_Comm_dup, MPI_Info_dup, \
+    from mpi4py.libmpi cimport MPI_Comm, MPI_Info, MPI_Comm_dup, MPI_Info_dup, \
                                MPI_Comm_free, MPI_Info_free
 
 # Initialization
