--- src/TKIVtkDraw/EXTERNLIB.orig	2016-09-22 08:57:41.308631725 +0200
+++ src/TKIVtkDraw/EXTERNLIB	2016-09-26 06:48:14.476393388 +0200
@@ -19,5 +19,5 @@
 vtkIOImage
 vtkImagingCore
 vtkInteractionStyle
-vtkRenderingOpenGL
-vtkRenderingFreeTypeOpenGL
+vtkRenderingOpenGL2
+vtkRenderingFreeType
--- src/TKIVtk/EXTERNLIB.orig	2016-09-22 08:57:53.148632469 +0200
+++ src/TKIVtk/EXTERNLIB	2016-09-26 06:47:53.939725440 +0200
@@ -12,8 +12,7 @@
 TKV3d
 vtkCommonCore
 vtkRenderingCore
-vtkRenderingOpenGL
+vtkRenderingOpenGL2
 vtkRenderingFreeType
-vtkRenderingFreeTypeOpenGL
 vtkFiltersGeneral
 vtkInteractionStyle
--- src/IVtkVTK/IVtkVTK_ShapeData.cxx.orig	2016-09-22 08:43:48.995246427 +0200
+++ src/IVtkVTK/IVtkVTK_ShapeData.cxx	2016-09-22 08:47:20.331926289 +0200
@@ -80,9 +80,9 @@
   vtkIdType aPointIdVTK = thePointId;
   myPolyData->InsertNextCell (VTK_VERTEX, 1, &aPointIdVTK);
   const vtkIdType aShapeIDVTK = theShapeID;
-  mySubShapeIDs->InsertNextTupleValue (&aShapeIDVTK);
+  mySubShapeIDs->InsertNextTypedTuple (&aShapeIDVTK);
   const vtkIdType aType = theMeshType;
-  myMeshTypes->InsertNextTupleValue (&aType);
+  myMeshTypes->InsertNextTypedTuple (&aType);
 }
 
 //================================================================
@@ -97,9 +97,9 @@
   vtkIdType aPoints[2] = { thePointId1, thePointId2 };
   myPolyData->InsertNextCell (VTK_LINE, 2, aPoints);
   const vtkIdType aShapeIDVTK = theShapeID;
-  mySubShapeIDs->InsertNextTupleValue (&aShapeIDVTK);
+  mySubShapeIDs->InsertNextTypedTuple (&aShapeIDVTK);
   const vtkIdType aType = theMeshType;
-  myMeshTypes->InsertNextTupleValue (&aType);
+  myMeshTypes->InsertNextTypedTuple (&aType);
 }
 
 //================================================================
@@ -124,9 +124,9 @@
 
     myPolyData->InsertNextCell (VTK_POLY_LINE, anIdList);
     const vtkIdType aShapeIDVTK = theShapeID;
-    mySubShapeIDs->InsertNextTupleValue (&aShapeIDVTK);
+    mySubShapeIDs->InsertNextTypedTuple (&aShapeIDVTK);
     const vtkIdType aType = theMeshType;
-    myMeshTypes->InsertNextTupleValue (&aType);
+    myMeshTypes->InsertNextTypedTuple (&aType);
     anIdList->Delete();
   }
 }
@@ -144,7 +144,7 @@
   vtkIdType aPoints[3] = { thePointId1, thePointId2, thePointId3 };
   myPolyData->InsertNextCell (VTK_TRIANGLE, 3, aPoints);
   const vtkIdType aShapeIDVTK = theShapeID;
-  mySubShapeIDs->InsertNextTupleValue (&aShapeIDVTK);
+  mySubShapeIDs->InsertNextTypedTuple (&aShapeIDVTK);
   const vtkIdType aType = theMeshType;
-  myMeshTypes->InsertNextTupleValue (&aType);
+  myMeshTypes->InsertNextTypedTuple (&aType);
 }
88a89
> #include <vtkVersion.h>
--- src/IVtkDraw/IVtkDraw.cxx.orig	2016-10-03 10:32:12.386093911 +0200
+++ src/IVtkDraw/IVtkDraw.cxx	2016-10-03 10:32:24.696094672 +0200
@@ -86,6 +86,7 @@
 #include <IVtkDraw_HighlightAndSelectionPipeline.hxx>
 #include <IVtkDraw_Interactor.hxx>
 
+#include <vtkVersion.h>
 #include <vtkAlgorithmOutput.h>
 #include <vtkAppendPolyData.h>
 #include <vtkBMPWriter.h>
