commit 23a5f890abfb3eb450065fff1fc75d8885b991cd
Author: Jefferson Amstutz <jefferson.d.amstutz@intel.com>
Date:   Thu Sep 1 13:52:28 2016 -0400

    fixes for Arch/gcc-6.1

diff --git a/apps/qtViewer/widgets/affineSpaceManipulator/QAffineSpaceManipulator.cpp b/apps/qtViewer/widgets/affineSpaceManipulator/QAffineSpaceManipulator.cpp
index e2ee89c..8098452 100644
--- a/apps/qtViewer/widgets/affineSpaceManipulator/QAffineSpaceManipulator.cpp
+++ b/apps/qtViewer/widgets/affineSpaceManipulator/QAffineSpaceManipulator.cpp
@@ -375,7 +375,9 @@ namespace ospray {
           for (int y=0;y<Ny;y++) {
             const float t = (y+0.f)/Ny*2.f*M_PI;
             const float f = (x+0.f)/Nx*2.f*M_PI;
-            vtx[x][y] = vec3f(cos(t)*sin(f),sin(t)*sin(f),cos(f));
+            vtx[x][y] = vec3f(ospcommon::cos(t)*ospcommon::sin(f),
+                              ospcommon::sin(t)*ospcommon::sin(f),
+                              ospcommon::cos(f));
             txt[x][y] = vec2f(x/float(Nx),y/float(Ny));
             vec3f c = (((x/4)+(y/4))%2) ? color_bright : color_dark;
             col[x][y] = vec3f(c);
diff --git a/ospray/lights/DirectionalLight.cpp b/ospray/lights/DirectionalLight.cpp
index b916aaf..13d101c 100644
--- a/ospray/lights/DirectionalLight.cpp
+++ b/ospray/lights/DirectionalLight.cpp
@@ -38,7 +38,7 @@ namespace ospray {
     direction = -normalize(direction); // the ispc::DirLight expects direction towards light source
 
     angularDiameter = clamp(angularDiameter, 0.f, 180.f);
-    const float cosAngle = cos(deg2rad(0.5f*angularDiameter));
+    const float cosAngle = ospcommon::cos(deg2rad(0.5f*angularDiameter));
 
     ispc::DirectionalLight_set(getIE(), (ispc::vec3f&)direction, (ispc::vec3f&)radiance, cosAngle);
   }
diff --git a/ospray/lights/SpotLight.cpp b/ospray/lights/SpotLight.cpp
index 030325d..e2bf17a 100644
--- a/ospray/lights/SpotLight.cpp
+++ b/ospray/lights/SpotLight.cpp
@@ -52,8 +52,8 @@ namespace ospray {
     direction = normalize(direction);
     openingAngle = clamp(openingAngle, 0.f, 180.f);
     penumbraAngle = clamp(penumbraAngle, 0.f, 0.5f*openingAngle);
-    const float cosAngleMax = cos(deg2rad(0.5f*openingAngle));
-    const float cosAngleMin = cos(deg2rad(0.5f*openingAngle - penumbraAngle));
+    const float cosAngleMax = ospcommon::cos(deg2rad(0.5f*openingAngle));
+    const float cosAngleMin = ospcommon::cos(deg2rad(0.5f*openingAngle - penumbraAngle));
     const float cosAngleScale = 1.0f/(cosAngleMin - cosAngleMax);
     
     ispc::SpotLight_set(getIE(),
